### PR TITLE
functionalizing asking of questions and checking of answers

### PR DESCRIPTION
**Pull request by gage-russell**
Moving the ask question and check answer functionality into their own functions.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -6,8 +6,9 @@
 from datetime import datetime
 import data
 import random
-from typing import Optional, List, Dict
+from typing import Optional, List, Dict, Union
 
+StrInt = Union[str, int]
 
 # NOTE-TO-SELF:
```
Comment (scott-sattler) [inline main.py:13]: can remove all notes - no longer applicable

File: main.py [hunk 2]
```diff
@@ -109,14 +98,47 @@ def administer_test(self):
                 else:
                     break
 
+    def _check_answer(self):
+        """
+        checks the user answer
+
+        :return: None
+        """
+        # TODO: remove use of str slicing and eval -- good for now, low priority, but too many ways this could go wrong
```
Comment (scott-sattler) [inline main.py:107]: I wonder who's idea eval() was lol

File: main.py [hunk 3]
```diff
@@ -6,8 +6,9 @@
 from datetime import datetime
 import data
 import random
-from typing import Optional, List, Dict
+from typing import Optional, List, Dict, Union
 
+StrInt = Union[str, int]
```
Comment (scott-sattler) [inline main.py:11]: what is this wizardry

File: main.py [hunk 4]
```diff
@@ -109,14 +98,47 @@ def administer_test(self):
...
+            self.correct += 1
+            print('correct', end=" || ")
+        else:
+            self.incorrect += 1
+            print(f'incorrect (the answer is {correct_answer})', end=" || ")
+        print(f'{self.correct} correct : {self.incorrect} incorrect')
+
+        return
+
+    def _ask_question(self):
```
Comment (scott-sattler) [inline main.py:119]: this is good

**Review by scott-sattler — APPROVED** [review]
good improvements!

File: main.py (hunk 3 above)
Comment (gage-russell) [reply main.py:11]: So `typing` is a python module for type-hinting.

You can use `typing.Union` for objects which could be more than one data type.

Essentially, I am creating a new data type (for type hinting only) which is either str or int.

**Pull request merged**